### DISCUSSION TITLE
keybind - sided modifier alone to trigger action (Quick Terminal) not working well

### DISCUSSION THREAD

**Discussion by edrozenberg** [ghostty-org/ghostty#3734] [Feature Requests, Ideas] [answered]
Ghostty Version 1.0.0
Build 8469
Commit 4b4d4062
(MacOS)

I looked into the possibility of using specifically a left or right modifier alone in a keybind to trigger an action such as Quick Terminal. For ex. to use the right side CTRL key to trigger Quick Terminal by itself, a single key as opposed to a combo.

Tried the following:
`keybind = ctrl+right_control=toggle_quick_terminal`

Results:
After initial Ghostty launch, it works well ONCE. Tap right CTRL to show Quick Terminal (works), tap it again to hide (works). Any subsequent taps of the right CTRL flicker the Quick Terminal on/off, or sometimes may have no noticeable effect at all. It works once after each Ghostty restart.

https://github.com/user-attachments/assets/def6141e-d20f-45ac-a729-484241dc687f

Not sure if this is even expected to work, but it would be nice esp. for keyboards that have all kinds of "extra modifier keys" to play with :).

This is only an issue with "sided modifiers"  such as left vs. right CTRL. Not an issue triggering with some other single key, for ex. the following works just fine:
`keybind = f10=toggle_quick_terminal`✅

**Comment by goooseman** [comment]
I also face this issue, my setup is `keybind = global:ctrl+`=toggle_quick_terminal`

For me it works the first time and then it works if I manually cmd+tab to the ghostty application to make it active. So

Following does not work:
1. Open quick terminal
2. Close it
3. Switch to Safari
4. Click ctrl+` <-- nothing happens

Following does not work:
1. Open quick terminal
2. Close it
3. Switch to Safari
4. Switch to Ghostty, there is no window, but I see Ghostty in the menu bar
5. Click ctrl+` - quick terminal opens


My config
```
   6   │ quick-terminal-position = bottom
   7   │ quick-terminal-screen = mouse
   8   │ quick-terminal-animation-duration = 0.2
   9   │ quick-terminal-autohide = false
  10   │
```

**Reply by edrozenberg** [reply]
I'm not able to replicate the issues you're seeing.

Try
`keybind = global:ctrl+grave_accent=toggle_quick_terminal`
(`grave_accent` instead of the actual character)

For `global` to work, make sure you have enabled Mac Settings -> Privacy & Security -> Accessibility -> Ghosstty added and enabled:

![Ghostty Acc](https://github.com/user-attachments/assets/81e520db-6422-4c4a-819e-d660c476fa57)

**Reply by goooseman** [reply]
Thx, looks like changing it to `grave_accent` has resolved the issue, works fine now

**Reply by kiru** [reply]
Thanks a lot. I had to untoggle and re-toggle after updating Ghostty to get the quick terminal working again.

**Comment by mitchellh** [comment, marked as answer]
#4335

**Comment by edrozenberg** [comment]
Closing as there is now an active issue to resolve this.
